build/storage/python_system_tools/test_platform.py

```diff
@@ -0,0 +1,262 @@
+import logging
+import re
+from concurrent.futures import ThreadPoolExecutor
+from typing import Optional
+
+from python_system_tools.terminal import Terminal
+from python_system_tools.consts import WORKSPACE_PATH, PMS
+
+
+class UnknownOSError(Exception):
+    """A custom Exception of a TestPlatform class"""
+
+
+class TestPlatform:
+    """
+    A class used to represent a testing platform
+
+    ...
+
+    Attributes
+    ----------
+    terminal: Terminal
+        Represents a session with an SSH server
+    os: str
+        The name of an operating system
+    pm: str
+        The name of a package manager
+
+    Methods
+    -------
+    check_virtualization()
+        Checks if VT-x/AMD-v support is enabled in BIOS
+    check_kvm()
+        Checks if kvm modules are loaded
+    check_system()
+        Checks the type of OS and returns it, raises UnknownOSError if it's unrecognized
+    install(program)
+        Installs a program on a machine
+    setup_docker_compose()
+        Checks if docker-compose is installed and installs it if it's not
+    setup_libguestfs_tools()
+        Installs libguestfs-tools for a specific OS
+    change_vmlinuz()
+        Changes the mode of /boot/vmlinuz-*
+    check_security_policies()
+        Checks if security policies are disabled
+    """
+
+    def __init__(self, terminal: Terminal):
+        """
+        Parameters
+        ----------
+        terminal: Terminal
+            A session with an SSH server
+        """
+
+        self.terminal = terminal
+        self.os = self.check_system()
+        self.pm = PMS[self.os]
+
+    def check_virtualization(self) -> bool:
+        """
+        Checks if VT-x/AMD-v support is enabled in BIOS
+
+        Returns
+        -------
+        bool
+            True if virtualization support is enabled in BIOS else False
+        """
+
+        logging.info("Checking if VT-x/AMD-v support is enabled in BIOS")
+        regex = re.compile(r"vt-x|amd-v|full", re.IGNORECASE)
+        out, _ = self.terminal.execute("lscpu | grep -i virtualization")
+        if regex.search(out):
+            logging.info("Virtualization support is enabled in BIOS")
+            return True
+        logging.error("Virtualization support is not enabled in BIOS")
```

Comment:
it's not error, it's normal situation that virtualization is not set

Reply:
changed to info